xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/flamingo.js

```diff
@@ -40,14 +40,90 @@ require(['jquery', 'bootstrap'], function($) {
   });
 });
+
+/*
+  Handle the behavior of all drawers on the page.
+  In order to create a drawer that is compatible with this script, two elements are necessary:
+  * A drawer toggler button, that will have:
+   ** a reference to the drawer container in its attribute 'aria-control'
+   ** the class 'drawer-toggle'
+  * A drawer container, that will have:
+   ** a unique ID
+   ** the class 'drawer-nav' (mostly for style) 
+   ** the class 'closed' if the drawer is supposed to start in a closed state.
+   ** a label 
+   ** The content to display inside. It's asserted that at least one element in this content can receive focus.
+  For an example of drawer creation, see #tmDrawerActivator and #tmDrawer.
+ */
+require(['jquery'], function($) {
+  // The overlay is the same whatever the drawer opened.
+  const drawerOverlay = $(".drawer-overlay");
+  $('.drawer-toggle').each(function(index) {
+    // Setting up the drawer.
+    let drawerContainerToggler = $(this);
+    let drawerId = drawerContainerToggler.attr('aria-controls');
+    let drawerContainer = $(document.getElementById(drawerId));
+    let focusableElements = drawerContainer.find('button, a, input:not([type="hidden"]), ' +
+     'select, textarea, [tabindex]:not([tabindex="-1"])');
+
+    // Note that the 'drawer-open' and 'drawer-close' CSS classes are added before the open and close animations end
+    // which prevents us from using them in automated tests. We need something more reliable so we listen to
+    // 'drawer1.opened' and 'drawer1.closed' events and add our own markers.
+    let openDrawer = () => drawerContainer.trigger('drawer' + index + '.opened');
+    let closeDrawer = () => drawerContainer.trigger('drawer' + index + '.closed');
+    drawerContainerToggler.on('click', openDrawer);
+    drawerOverlay.on('click', closeDrawer);
+
+    drawerContainer.on('drawer' + index + '.opened', function(event) {
+      drawerContainerToggler.attr('aria-expanded', 'true');
+      // We update the state of the drawer (using setAttribute since it's faster)
+      drawerContainer.get(0).setAttribute('open','open');
+      function focusDrawer() {
+        if (drawerContainer.get(0).hasAttribute('open')) {
+          drawerContainer.trigger('focus');
+        }
+      };
+      function focusLastItem() {
+        if (drawerContainer.get(0).hasAttribute('open') && focusableElements.length !== 0) {
+          focusableElements.last().trigger('focus');
+        }
+      };
+      // The drawer can be closed by pressing the ESC key.
+      $("body").on('keydown.drawerClose', function (event) {
+        if (event.key === 'Escape') {
+          closeDrawer();
+        }
+      });
+      // The focus is set back to the start by setting focus outside of it
+      focusableElements.on('blur.drawerClose', function (event) {
+        if (event.relatedTarget != null && event.relatedTarget.closest('#' + $.escapeSelector(drawerId)) == null) {
+          focusDrawer();
```

Comment:
Why is this necessary? The dialog element should have native focus handling.

Reply:
Without using .showModal, it did not handle it natively ... 
Addressed in https://github.com/xwiki/xwiki-platform/pull/2282/commits/8f6b02b56dcd0bde6c9f0fe4a568a34e4c976feb